tools/pyboard.py

```diff
@@ -370,7 +406,24 @@ def exec_raw_no_follow(self, command):
         if not data.endswith(b">"):
             raise PyboardError("could not enter raw repl")
 
-        # write command
+        if self.use_raw_paste:
+            # Try to enter raw-paste mode.
+            self.serial.write(b"\x05A\x01")
+            data = self.serial.read(2)
+            if data == b"R\x00":
+                # Device understood raw-paste command but doesn't support it.
+                pass
+            elif data == b"R\x01":
+                # Device supports raw-paste mode, write out the command using this mode.
+                return self.raw_paste_write(command_bytes)
+            else:
+                # Device doesn't support raw-paste, fall back to normal raw REPL.
```

Comment:
Should this then set `self.use_raw_paste = False` ?

Reply:
Yeah that's probably a good idea, so it doesn't keep trying the failing raw-paste each time.  And it's not like the device can suddenly support it, you'd have to at least disconnect/reconnect to get support for raw-paste, which means resetting this variable.

Reply:
done